applications/DEMApplication/custom_constitutive/DEM_rolling_friction_model_constant_torque.cpp

```diff
@@ -29,56 +32,50 @@ namespace Kratos{
         
     }
 
-    void DEMRollingFrictionModelConstantTorque::ComputeRollingFriction(SphericParticle* p_element, SphericParticle* p_neighbor, double LocalContactForce[3], array_1d<double, 3>& mContactMoment, double indentation)
+    void DEMRollingFrictionModelConstantTorque::ComputeRollingFriction(SphericParticle* p_element, SphericParticle* p_neighbor, double LocalCoordSystem_2[3], double LocalContactForce[3], array_1d<double, 3>& mContactMoment)
     {
         array_1d<double, 3> elementRelAngularVelocity;
         noalias(elementRelAngularVelocity) = p_element->GetGeometry()[0].FastGetSolutionStepValue(ANGULAR_VELOCITY) - p_neighbor->GetGeometry()[0].FastGetSolutionStepValue(ANGULAR_VELOCITY);
         if (elementRelAngularVelocity[0] || elementRelAngularVelocity[1] || elementRelAngularVelocity[2]){
-            const double my_young      = p_element->GetYoung(); 
-            const double other_young   = p_neighbor->GetYoung();
-            const double my_arm_length = p_element->GetInteractionRadius() - indentation * other_young / (other_young + my_young);
-            //TODO: This should be updated for non-spherical particle
-            double contact_center_point_to_element1_mass_center_distance = my_arm_length; //Here, this only works for sphere particles
-                       
+            // Normalize relative angular velocity
             GeometryFunctions::normalize(elementRelAngularVelocity);
-
             array_1d<double, 3> elementRelAngularVelocity_normalise = elementRelAngularVelocity;
 
-            Properties& properties_of_this_contact = p_element->GetProperties().GetSubProperties(p_neighbor->GetProperties().Id());
-
-            mContactMoment[0] -= elementRelAngularVelocity_normalise[0] * fabs(LocalContactForce[2]) * contact_center_point_to_element1_mass_center_distance * properties_of_this_contact[ROLLING_FRICTION]; 
-
-            mContactMoment[1] -= elementRelAngularVelocity_normalise[1] * fabs(LocalContactForce[2]) * contact_center_point_to_element1_mass_center_distance * properties_of_this_contact[ROLLING_FRICTION]; 
+            // Get equivalent radius (this only works for sphere particles)
+            const double my_radius    = p_element->GetRadius();
+            const double other_radius = p_neighbor->GetRadius();
+            const double equiv_radius = my_radius * other_radius / (my_radius + other_radius);
```

Comment:
For the equiv_radius, although we have discussed about it, I believe the Jiang's version could be better than this version. Because we didn't use this equiv_radius for wall. What's more, if the arm_length (equiv_radius) is not related to energy conservation, the old version could be better.

Reply:
"For the equiv_radius, although we have discussed about it, I believe the Jiang's version could be better than this version."
What is the Jiang's version?

"Because we didn't use this equiv_radius for wall."
Actually, I am using the equiv_radius for the particle-wall contact, because the result of the equiv_radius formula when you use an infinite radius for the the other particle (i.e. the wall) is simply the radius of the particle.

"What's more, if the arm_length (equiv_radius) is not related to energy conservation, the old version could be better."
I didn't understand it well.
But, as I said, my concern of taking into account only the radius of one of the interacting particle in the rolling friction formula is that the rolling friction value will be different for each particle, which may affect the energy conservation. Thus, I would use the same radius in the formula of both particles, be it the effective radius, the minimum radius, or any other option.